server/src/main/java/org/elasticsearch/action/admin/indices/diskusage/IndexDiskUsageAnalyzer.java

```diff
@@ -0,0 +1,665 @@
+                int docID = 0;
+                for (long idx = 0; idx <= 8; idx++) {
+                    final int skipDocID = Math.toIntExact(idx * reader.maxDoc() / 8);
+                    if (skipDocID <= docID) {
+                        cancellationChecker.logEvent();
+                        postings = termsEnum.postings(postings, PostingsEnum.NONE);
+                        if ((docID = postings.advance(skipDocID)) != DocIdSetIterator.NO_MORE_DOCS) {
+                            postings.freq();
+                            postings.nextDoc();
+                        }
+                    }
+                }
+            }
+            termsIndexLookup.seekExact(terms.getMax());
+            final long termsBytes = directory.getBytesRead();
+            stats.addTerms(field.name, termsBytes);
+
+            // Visit positions, offsets, and payloads
+            if (field.getIndexOptions() == IndexOptions.DOCS_AND_FREQS_AND_POSITIONS ||
+                field.getIndexOptions() == IndexOptions.DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS) {
+                termsEnum = terms.iterator();
+                while (termsEnum.next() != null) {
+                    cancellationChecker.logEvent();
+                    termsEnum.docFreq();
+                    postings = termsEnum.postings(postings, PostingsEnum.ALL);
+                    while (postings.nextDoc() != DocIdSetIterator.NO_MORE_DOCS) {
+                        if (terms.hasPositions()) {
+                            for (int pos = 0; pos < postings.freq(); pos++) {
+                                postings.nextPosition();
+                                postings.startOffset();
+                                postings.endOffset();
+                                postings.getPayload();
+                            }
+                        }
+                    }
+                }
+                final long proximityBytes = directory.getBytesRead() - termsBytes;
+                stats.addProximity(field.name, proximityBytes);
+            }
+        }
+    }
+
+    void analyzePoints(SegmentReader reader, IndexDiskUsageStats stats) throws IOException {
+        PointsReader pointsReader = reader.getPointsReader();
+        if (pointsReader == null) {
+            return;
+        }
+        pointsReader = pointsReader.getMergeInstance();
+        final PointValues.IntersectVisitor visitor = new PointValues.IntersectVisitor() {
+            @Override
+            public void visit(int docID) {
+                assert false : "Must never be called";
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public void visit(int docID, byte[] packedValue) {
+                cancellationChecker.logEvent();
+            }
+
+            @Override
+            public PointValues.Relation compare(byte[] minPackedValue, byte[] maxPackedValue) {
+                return PointValues.Relation.CELL_CROSSES_QUERY;
```

Comment:
One hack we could do to save time would be to return `CELL_OUTSIDE_QUERY` for all cells that are strictly within the global boundaries (`pointValues.getMinPackedValue()` and `pointValues.getMaxPackedValue()`).

Reply:
Yes, I pushed https://github.com/elastic/elasticsearch/pull/74051/commits/242ded17e9bf1f83cd46539436c2f50d8cb5673f to look up the min and the max points only.